Assets/_BForBoss/_Character/Scripts/DashLinesEffectBehaviour.cs

```diff
@@ -0,0 +1,56 @@
+//-----------------------------------------------------------------------
+// <copyright file="DashLinesEffectBehaviour.cs" company="PERIGON GAMES">
+//     Copyright (c) PERIGON GAMES. 2020 All Rights Reserved
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Perigon.Character
+{
+    public class DashLinesEffectBehaviour : MonoBehaviour
```

Comment:
So below you're trying to `GetComponent<RawImage>()` But there's always that change that there isn't a RawImage on the same GameObject.  Which would then throw a nullpointer in `SetFloat`

However, if you write `[RequireComponent(typeof(RawImage))]` right above the `public class....` line, it should guarantee whoever is using this script that it must have a RawImage. So Unity does the null check for you.